pupil-spa/src/app/aa-colours/aa-colours.component.ts

```diff
@@ -42,6 +45,22 @@ export class AAColoursComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
+  @HostListener('window:beforeunload', ['$event'])
+  unloadNotification() {
+    this.auditService.addEntry(new RefreshDetected());
```

Comment:
This will be executed on both page refresh and tab close action.
I am considering replacing this with a new `AppUnload` event to describe it more accurately.
Looking forward to your thoughts

Reply:
i think `AppUnload` is rather vague, and it should be more browser action specific.  How about `RefreshOrTabClose`? at least something that tells me exactly what the user has done

Reply:
Renamed to `RefreshOrTabCloseDetected`

Reply:
Thanks George 🏆